main_svc

Tor1ant opened the pull request:
main_server реализован

Tor1ant committed:
/admin/users - реализовано

Tor1ant committed:
/admin/users - реализовано

Tor1ant committed:
/admin/categories - реализовано
/categories - реализовано

Tor1ant committed:
Merge branch 'categories' into main_svc

# Conflicts:
#	main-server/src/main/java/com/github/explore_with_me/main/exception/handler/ExceptionAdviceHandler.java
#	main-server/src/main/java/com/github/explore_with_me/main/user/controller/AdminUserController.java
#	main-server/src/main/java/com/github/explore_with_me/main/user/model/User.java
#	main-server/src/main/resources/schema.sql

Tor1ant committed:
main-server реализован

Alexander-Bolotov commented on main-server/src/main/java/com/github/explore_with_me/main/user/controller/paramEntity/UsersParam.java:
```diff
@@ -0,0 +1,14 @@
+
+@AllArgsConstructor
+@Getter
+public class UsersParam {
```
этот класс не нужен. Используй параметры отдельно, как они есть

Tor1ant replied:
исправлено.

Alexander-Bolotov commented on main-server/src/main/java/com/github/explore_with_me/main/user/dto/NewUserDto.java:
```diff
@@ -0,0 +1,18 @@
+import lombok.Data;
+
+@Data
+public class NewUserDto {
```
Используй UserDto - просто будешь присылать ее без id

Tor1ant replied:
исправлено.

Alexander-Bolotov commented on main-server/src/main/java/com/github/explore_with_me/main/user/dto/UserShortDto.java:
```diff
@@ -0,0 +1,10 @@
+import lombok.Data;
+
+@Data
+public class UserShortDto {
```
тоже можно использовать UserDto

Tor1ant replied:
исправлено

Alexander-Bolotov commented on main-server/src/main/java/com/github/explore_with_me/main/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,59 @@
+        } catch (Exception e) {
+            StringBuilder stringBuilder = new StringBuilder(e.getCause().getCause().getMessage());
+            int indexEqualsSign = stringBuilder.indexOf("=");
+            stringBuilder.delete(0, indexEqualsSign + 1);
```
не совсем понимаю, что происходит в этих строках. Думаю просто можно передать в сообщении, что "произошла ошибка при создании пользователя с таким то id и добавить причину"

Alexander-Bolotov replied:
и тоже в лог записать

Tor1ant replied:
Это обработка исключения, которое выбрасывается базой данных. Я беру самое нижнее по вложенности исключение, беру его сообщение и обрабатываю стрингбилдером, чтобы передать на фронтенд "красивое" сообщение. Точно стоит отказаться от этого?

Alexander-Bolotov replied:
Если эта обработка подходит для любого исключения, а не только конкретно для этого, то вынеси в отдельный служебный метод и вызывай его когда требуется

Tor1ant replied:
> Если эта обработка подходит для любого исключения, а не только конкретно для этого, то вынеси в отдельный служебный метод и вызывай его когда требуется

Такой еще вопрос. Можно ли подобную обработку вообще вынести в ExceptionHandler и сделать этот метод статическим, чтобы я мог вызывать его вообще во всем коде или так делать не стоит? Если так делать можно, то я во время третьего этапа так и сделаю

Alexander-Bolotov commented on main-server/src/main/java/com/github/explore_with_me/main/paramEntity/EventFindParam.java:
```diff
@@ -0,0 +1,20 @@
+@EqualsAndHashCode
+@Getter
+@RequiredArgsConstructor
+public class EventFindParam {
```
тоже не вижу в необходимости этих классов, используй параметры отдельно, как есть

Tor1ant replied:
исправлено. Нам просто на одном вебинаре показали, что так можно делать для удобства.

Tor1ant committed:
Удаление избыточных классов параметров

Удалены классы сущностей параметров, такие как EventFindParam, PaginationParams и GetEventsParam, и заменены на прямые параметры в методах сервиса. Это упрощает код за счет удаления ненужных классов и делает методы более явными в отношении необходимых им параметров. Это также повышает читабельность кода и делает вызовы методов менее громоздкими.
